### DISCUSSION TITLE
Give access to base fonts

### DISCUSSION THREAD

**Discussion by Xyndra** [korlibs/korge#2019] [Ideas and discussions]
Every system has a set of base fonts, at least one normal and one monospace. It would be a very nice feature to be able to use these instead of having to implement them manually.

**Comment by soywiz** [comment]
You can list system fonts like this:
https://github.com/korlibs/korge/blob/6b1313368e63cde76827c4d9809bee6e5984dda9/korge-core/src/common/korlibs/image/font/SystemFont.kt#L16-L19

**Reply by Xyndra** [reply]
That might create problems when working with different platforms though. I am thinking more like the css way.

**Reply by soywiz** [reply]
Why? What's the css way you have in mind? Can you propose an API that would work for you?

**Reply by Xyndra** [reply]
I propose this way: https://www.w3schools.com/css/tryit.asp?filename=trycss_font-family

**Reply by soywiz** [reply]
What about:

```kotlin
suspend fun getFirstFont(vararg fontNames: String): Font?
```

Usage:

```kotlin
SystemFont.getFirstFont("Times New Roman", "Times", "serif", "sans-serif", "monospace")
```

It could match the font name ignoring cases, then could try to match family name like serif, sans-serif or monospace.
?

**Reply by soywiz** [reply]
I have created this issue: https://github.com/korlibs/korge/issues/2047

So anyone can grab it and implement it.
